### ISSUE TITLE
As a plugin writer, I have to opt-out to having a "non-redirect" behavior in the content app

### ISSUE THREAD

**Issue by bmbouter** [pulp/pulpcore#3459]
With the pulpcore 3.22 release, we planned to make the "redirect to include a trailing slash" behavior a plugin opt-in feature. We couldn't make it opt-out because pulp_container breaks with this and pre-existing GA releases of pulp_container declare forward compatibility through pulpcore 3.24.

This issue tracks the change:  With 3.25, have the "redirect to include a trailing slash" be a plugin opt-out feature and have it enabled by default.

**Comment by bmbouter** [comment]
@pulp/core FYI

**Comment by daviddavis** [comment]
Just adding some historical notes here.

The original redirect feature issue: https://github.com/pulp/pulpcore/issues/3173

The original PR to add the feature: https://github.com/pulp/pulpcore/pull/3434

The PR that reverted/removed the feature: https://github.com/pulp/pulpcore/pull/3465

**Comment by ggainey** [comment]
Clarification: implementing this requires more than just re-pushing the PR - it requires adding machinery to let plugins opt in/out of the behavior

**Comment by lubosmj** [comment]
I do not understand the need to enable or disable the feature in the plugin's code. We decided to revert the commit because we had compatibility issues in pulp_container (https://github.com/pulp/pulpcore/pull/3434#discussion_r1047175632) and pulp_python (https://github.com/pulp/pulpcore/pull/3456). The branch where the commit was merged had not been identified as a branch holding breaking changes.

Is it acceptable to merge the very same commit with, for instance, a hook method that will enable plugin writers to create an additional wrap-around on top of the non-redirect workflow, if needed? This is what we were trying to add to pulp_container to fix the issues: https://github.com/pulp/pulp_container/pull/1179.
